game/controllers-midi.cc

```diff
@@ -3,22 +3,22 @@
 #include "controllers.hh"
 #include "portmidi.hh"
 #include "fs.hh"
-#include <boost/regex.hpp>
+#include <regex>
 #include <unordered_map>
 
 namespace input {
 
 	class Midi: public Hardware {
 	public:
 		Midi() {
-			boost::regex re(config["game/midi_input"].s());
+			std::regex re(config["game/midi_input"].s());
 			for (int dev = 0; dev < Pm_CountDevices(); ++dev) {
 				try {
 					PmDeviceInfo const* info = Pm_GetDeviceInfo(dev);
 					if (!info->input) continue;  // Not an input device
 					if (info->opened) continue;  // Already opened
 					std::string name = getName(dev);
-					if (!re.empty() && !regex_search(name, re)) continue;
+					if (!config["game/midi_input"].s().empty() && !regex_search(name, re)) continue;
```

Comment:
If `config["game/midi_input"].s()` returns an empty string, than the check `!config["game/midi_input"].s().empty()` is not necesairy since `std::regex re` may be initialized with an empty string and than matching nothing. This goes for all these 'empty' checks

Reply:
Nice work on getting it all replaced though! :+1: